### PR TITLE
[bitnami/nginx] Set /app as safe directory when enabling gitSync

### PR DESCRIPTION
**Pull request by TheiLLeniumStudios**
### Description of the change

Added `/app` as a safe directory to global git config as newer versions of git require you to do that

### Applicable issues

- Fixes https://github.com/bitnami/charts/issues/20682

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/nginx/templates/deployment.yaml [hunk 1]
```diff
@@ -131,6 +131,7 @@ spec:
             - -ec
             - |
               [[ -f "/opt/bitnami/scripts/git/entrypoint.sh" ]] && source "/opt/bitnami/scripts/git/entrypoint.sh"
+              git config --global --add safe.directory /app
               while true; do
                   cd /app && git pull origin {{ .Values.cloneStaticSiteFromGit.branch }}
```
Comment (andresbono) [inline bitnami/nginx/templates/deployment.yaml:136]: I would avoid having to use the `git config --global` command as it tries to persist the configuration in `/.gitconfig` and that file may not be writable by an unprivileged user. I see it was already mentioned in the associated issue https://github.com/bitnami/charts/issues/20682#issuecomment-1844993985, and to workaround that, you are setting `HOME=/tmp`, but that is not set by default, right?

What if we just do the following instead?

```suggestion
              while true; do
                  cd /app && git -c safe.directory=/app pull origin {{ .Values.cloneStaticSiteFromGit.branch }}
```

**Review by andresbono — COMMENTED** [review]
Thank you for your contribution! Could you please check my suggestion about your fix and let me know what do you think?

**Comment by TheiLLeniumStudios** [comment]
@andresbono accepted your change. Can we get this merged now please?

**Comment by johanneskastl** [comment]
Any news on this?

**Review by andresbono — APPROVED** [review]
LGTM, Thank you for your contribution!

**Pull request merged**